src/main/scala/scynamo/ScynamoEncoder.scala

```diff
@@ -12,119 +12,150 @@ import shapeless.tag.@@
+    numberStringEncoder.contramap(_.toEpochMilli.toString)
 
   implicit val instantTtlEncoder: ScynamoEncoder[Instant @@ TimeToLive] =
     numberStringEncoder.contramap[Instant @@ TimeToLive](_.getEpochSecond.toString)
 
-  implicit val uuidEncoder: ScynamoEncoder[UUID] = stringEncoder.contramap[UUID](_.toString)
+  implicit val uuidEncoder: ScynamoEncoder[UUID] =
+    stringEncoder.contramap(_.toString)
 
-  implicit def seqEncoder[A: ScynamoEncoder]: ScynamoEncoder[scala.collection.immutable.Seq[A]] =
-    value => value.toVector.parTraverse(ScynamoEncoder[A].encode).map(xs => AttributeValue.builder().l(xs: _*).build())
+  implicit def seqEncoder[A](implicit element: ScynamoEncoder[A]): ScynamoEncoder[Seq[A]] =
+    ScynamoEncoder.instance { xs =>
+      var allErrors  = Chain.empty[ScynamoEncodeError]
+      val attrValues = List.newBuilder[AttributeValue]
+      for ((x, i) <- xs.iterator.zipWithIndex) element.encode(x) match {
+        case Right(attr)  => attrValues += attr
+        case Left(errors) => allErrors ++= StackFrame.encoding(errors, Index(i)).toChain
+      }
+
+      NonEmptyChain.fromChain(allErrors).toLeft(AttributeValue.builder.l(attrValues.result(): _*).build())
+    }
 
   implicit def listEncoder[A: ScynamoEncoder]: ScynamoEncoder[List[A]] =
-    value =>
-      value.zipWithIndex
-        .parTraverse { case (x, i) =>
-          ScynamoEncoder[A].encode(x).leftMap(_.map(_.push(Index(i))))
-        }
-        .map(xs => AttributeValue.builder().l(xs: _*).build())
+    seqEncoder[A].narrow
 
   implicit def vectorEncoder[A: ScynamoEncoder]: ScynamoEncoder[Vector[A]] =
-    value =>
-      value.zipWithIndex
-        .parTraverse { case (x, i) =>
-          ScynamoEncoder[A].encode(x).leftMap(_.map(_.push(Index(i))))
-        }
-        .map(xs => AttributeValue.builder().l(xs: _*).build())
+    seqEncoder[A].narrow
 
-  implicit def setEncoder[A: ScynamoEncoder]: ScynamoEncoder[Set[A]] = listEncoder[A].contramap[Set[A]](x => x.toList)
+  implicit def setEncoder[A: ScynamoEncoder]: ScynamoEncoder[Set[A]] =
+    listEncoder[A].contramap(_.toList)
 
-  implicit def optionEncoder[A: ScynamoEncoder]: ScynamoEncoder[Option[A]] = {
-    case Some(value) => ScynamoEncoder[A].encode(value)
-    case None        => Right(AttributeValue.builder().nul(true).build())
-  }
+  implicit def optionEncoder[A](implicit element: ScynamoEncoder[A]): ScynamoEncoder[Option[A]] =
+    ScynamoEncoder.instance {
+      case Some(value) => element.encode(value)
+      case None        => rightNul
+    }
 
-  implicit def someEncoder[A: ScynamoEncoder]: ScynamoEncoder[Some[A]] = x => ScynamoEncoder[A].encode(x.get)
+  implicit def someEncoder[A](implicit element: ScynamoEncoder[A]): ScynamoEncoder[Some[A]] =
+    ScynamoEncoder.instance(some => element.encode(some.get))
 
-  implicit val finiteDurationEncoder: ScynamoEncoder[FiniteDuration] = longEncoder.contramap(_.toNanos)
+  implicit val finiteDurationEncoder: ScynamoEncoder[FiniteDuration] =
```

Comment:
This is a breaking change I think :warning: .  Before this, we saved FiniteDuration as type Long, now it would be of type String in Dynamo.

Reply:
`longEncoder` is just `numberStringEncoder.contramap(_.toString)`

Reply:
Oh okay nevermind then ;)